### ISSUE TITLE
DELETE fails when using subqueries and partitioned tables

### ISSUE THREAD

**Issue by volkovandr** [greenplum-db/gpdb#1104]
Hello, 

Compiled Greenplum from source code, last commit 14df7740c81e668ec54151d314a4528d60222660

Trying to execute a delete statement and it fails with "Unexpected internal error (planmain.c:275)"
Here is an example:

```
psql (9.5.3, server 8.3.23)
You are now connected to database "postgres" as user "gpadmin".
postgres=# create database db;
CREATE DATABASE
postgres=#
postgres=# \c db
psql (9.5.3, server 8.3.23)
You are now connected to database "db" as user "gpadmin".
db=#
db=# CREATE TABLE a
db-# (
db(#   transaction_id text not null,
db(#   event_id text not null,
db(#   issue_date timestamptz not null default now()
db(# )
db-# distributed by (transaction_id)
db-# partition by range (issue_date)
db-# (
db(#   start (timestamp without time zone '2016-08-01') INCLUSIVE
db(#   end (timestamp without time zone '2016-10-01') EXCLUSIVE
db(#   every (interval '1 month')
db(# );
NOTICE:  CREATE TABLE will create partition "a_1_prt_1" for table "a"
NOTICE:  CREATE TABLE will create partition "a_1_prt_2" for table "a"
CREATE TABLE
db=#
db=# CREATE TABLE b
db-# (
db(#   transaction_id text not null,
db(#   event_id text not null,
db(#   issue_date timestamp without time zone not null default now()
db(# )
db-# distributed by (transaction_id)
db-# partition by range (issue_date)
db-# (
db(#   start (timestamp without time zone '2016-08-01') INCLUSIVE
db(#   end (timestamp without time zone '2016-10-01') EXCLUSIVE
db(#   every (interval '1 month')
db(# );
NOTICE:  CREATE TABLE will create partition "b_1_prt_1" for table "b"
NOTICE:  CREATE TABLE will create partition "b_1_prt_2" for table "b"
CREATE TABLE
db=#
db=# delete from b where (transaction_id, event_id) in (select transaction_id, event_id from a);
ERROR:  Unexpected internal error (planmain.c:275)
```

Tried on two different clusters with different number of nodes with the same result.
The query fails in both cases when GPORCA is enabled or disabled.
The query works well if at least one of the tables is not partitioned. 
And I know for sure this query worked in a database compiled in April 2016.

**Comment by volkovandr** [comment]
However this works well:
```
db=# delete from b using a where b.transaction_id = a.transaction_id and b.event_id = a.event_id;
DELETE 0
```

**Comment by vraghavan78** [comment]
Thanks for reporting this. We will take a look at this.

**Comment by micklf** [comment]
same problem

**Comment by micklf** [comment]
same problem to me in legacy optimizer, any idea?

**Comment by hlinnaka** [comment]
This was fixed in GPDB version 5.10.0, by commit e5c9cfecd0a2dec19dc749c8c635f92d08c7f18a. Closing. Sorry for the delay!

The bug was in the Planner, but you got the same error with `optimizer=on`, because GPORCA falls back to the Planner with this query.
